roles/clean-devstack-default-resources/tasks/main.yaml

```diff
@@ -17,5 +17,14 @@
       openstack subnet delete public-subnet
       openstack subnet delete private-subnet
       openstack network delete private public
-      openstack flavor delete `openstack flavor list -f value -c ID --sort-column ID | tail -n +6` || true
+      flavor_white_list="'m1.tiny' 'm1.small' 'm1.medium' 'm1.large' 'm1.xlarge'"
+      for flavor_name in `openstack flavor list -f value -c Name`
+      do
+          if [[ "${flavor_white_list}" =~ "${flavor_name}" ]]; then
```

Comment:
I just tried to find a simple way:
```
flavors=`echo $(openstack flavor list -f value -c ID) | python -c "import sys; print ' '.join(list(set(sys.stdin.readline().split())-set(['1','2','3','4','5'])))"`
openstack flavor delete $flavors
```

But this is OK if you like this.

Reply:
Thank you sharing this way, but I want to keep simple and readable, I think my way is more easy to understand.